index.js

```diff
@@ -10,10 +10,17 @@
 var typeOf = require('kind-of');
 
 module.exports = function isNumber(num) {
+  if (!num && num !== 0) return false;
```

Comment:
Return earlier concept. Not sure enough but yea as i said, micro optimization - it just won't hit the kind-of if it is null or undef. It makes little sense to have it.